### PR TITLE
Move Reanimated Babel Plugin dependencies to dependencies

### PR DESCRIPTION
**Pull request by tjzel**
## Summary

After thinking about this, there's no reason for us to keep up the weird situation, with actually used Babel plugins in peer dependencies.

Reasons for that are:
- those plugins are supposed to perform a given transformation, regardless of their version and whether they are proposals or not. 
- React Native is currently in progress of changing those proposal plugins to accepted ones, it would give it unnecessary maintenance to check which are available.

## Test plan

I checked that our Example app works properly, there are no differences in Jest tests and also Reanimated works with Expensify's app.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,31 +91,23 @@
   },
   "homepage": "https://github.com/software-mansion/react-native-reanimated#readme",
   "dependencies": {
-    "@babel/plugin-transform-object-assign": "^7.16.7",
+    "@babel/plugin-transform-arrow-functions": "^7.23.3",
+    "@babel/plugin-transform-nullish-coalescing-operator": "^7.23.4",
+    "@babel/plugin-transform-optional-chaining": "^7.23.4",
+    "@babel/plugin-transform-shorthand-properties": "^7.23.3",
+    "@babel/plugin-transform-template-literals": "^7.23.3",
```
Comment (tomekzaw) [inline package.json:98]: Can we use `^7.0.0-0` for all of them?

Comment (tjzel) [reply package.json:98]: Smart, will do.

**Review by piaskowyk — APPROVED** [review]
Good for me 🫡

**Review by tomekzaw — APPROVED** [review]
Let it be

**Pull request merged**